test/e2e/create_test.go

```diff
@@ -173,12 +173,55 @@ func (tc *testContext) testMachineConfiguration(t *testing.T) {
 		// Replace the known private key with a randomly generated one.
 		err = tc.createPrivateKeySecret(false)
 		require.NoError(t, err, "error replacing private key secret")
+		// Wait for machines to be reconciled by deletion
```

Comment:
Thanks for putting in the work to identify this issue, and come up with this solution.
Having not thought of this test code in a while, I had to think through what was going on here.
This is testing the case that
* a Machine is provisioned with an outdated private key, 
* That machine goes into the provisioned state
* The Machine is not configured into a Node
* Change the key, and check that the provisioned Machine is deleted

What I'm not able to figure out is how this functionality is accomplished, I could not find the logic in WMCO which handles the deletion of a Machine without a Node object.  Could you please help me out here and point me to that?

Additionally the amount of logic here makes me believe that if this testing is required this should be a separate test instead of being shoehorned into the Machine Node configuration test.

Reply:
I agree with @sebsoto. I don't see how the Machines are deleted if they have not been configured into Nodes before?

Reply:
Are we hitting an auth error and deleting the Machine? [Ref](https://github.com/openshift/windows-machine-config-operator/blob/f94689375166cbfdcf35959f2dfa2dd6cdbcc419/controllers/windowsmachine_controller.go#L355)

Reply:
> Additionally the amount of logic here makes me believe that if this testing is required this should be a separate test instead of being shoehorned into the Machine Node configuration test.

True.

Reply:
> What I'm not able to figure out is how this functionality is accomplished, I could not find the logic in WMCO which handles the deletion of a Machine without a Node object. Could you please help me out here and point me to that?


@sebsoto  As you mentioned, the Machine is provisioned with an outdated private key hence WMCO is not able to establish a SSH connection; the reconciliation outcome is to re-create the machine. [See the code](https://github.com/openshift/windows-machine-config-operator/blob/cec3f910f4bd041003e768ccb17c7a712fdd4d1a/controllers/windowsmachine_controller.go#L347-L355) in WMCO responsible for that and below a snippet of the log from a dev cluster. 

```
2023-09-14T23:45:41Z	DEBUG	controller.windowsmachine	invalid Machine	{"name": "e2e-wm-xx85w", "error": "no IP addresses defined"}
2023-09-14T23:45:42Z	DEBUG	controller.windowsmachine	invalid Machine	{"name": "e2e-wm-xx85w", "error": "no IP addresses defined"}
2023-09-14T23:45:42Z	DEBUG	controller.windowsmachine	reconciling	{"windowsmachine": {"Name":"e2e-wm-xx85w","Namespace":"openshift-machine-api"}}
2023-09-14T23:45:54Z	INFO	controller.secret	updating secret	{"name": "windows-user-data"}
2023-09-14T23:45:54Z	DEBUG	controller.windowsmachine	reconciling	{"windowsmachine": {"Name":"e2e-wm-xx85w","Namespace":"openshift-machine-api"}}
2023-09-14T23:45:54Z	INFO	controller.windowsmachine	processing	{"windowsmachine": {"Name":"e2e-wm-xx85w","Namespace":"openshift-machine-api"}, "address": "10.0.128.7"}
2023-09-14T23:45:54Z	DEBUG	wc 10.0.128.7	initializing SSH connection
2023-09-14T23:48:04Z	DEBUG	wc 10.0.128.7	SSH dial	{"IP Address": "10.0.128.7", "error": "dial tcp 10.0.128.7:22: connect: connection timed out"}
2023-09-14T23:49:04Z	DEBUG	wc 10.0.128.7	SSH dial	{"IP Address": "10.0.128.7", "error": "ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain"}
2023-09-14T23:49:04Z	DEBUG	events	Machine e2e-wm-xx85w authentication failure	{"type": "Warning", "object": {"kind":"Machine","namespace":"openshift-machine-api","name":"e2e-wm-xx85w","uid":"fc8f48c4-ccfb-408b-96d6-23648268cd99","apiVersion":"machine.openshift.io/v1beta1","resourceVersion":"72990"}, "reason": "MachineSetupFailure"}
2023-09-14T23:49:04Z	INFO	controller.windowsmachine	machine has been remediated by deletion	{"name": "e2e-wm-xx85w"}
2023-09-14T23:49:04Z	DEBUG	events	Machine e2e-wm-xx85w has been remediated by deleting the Machine object	{"type": "Normal", "object": {"kind":"Machine","namespace":"openshift-machine-api","name":"e2e-wm-xx85w","uid":"fc8f48c4-ccfb-408b-96d6-23648268cd99","apiVersion":"machine.openshift.io/v1beta1","resourceVersion":"72990"}, "reason": "MachineDeleted"}
2023-09-14T23:49:04Z	DEBUG	controller.windowsmachine	Machine has no phase associated with it	{"name": "e2e-wm-2p7gp"}
2023-09-14T23:49:05Z	DEBUG	controller.windowsmachine	Machine has no phase associated with it	{"name": "e2e-wm-2p7gp"}
2023-09-14T23:49:06Z	DEBUG	controller.windowsmachine	invalid Machine	{"name": "e2e-wm-2p7gp", "error": "no IP addresses defined"}
2023-09-14T23:49:08Z	DEBUG	controller.windowsmachine	invalid Machine	{"name": "e2e-wm-2p7gp", "error": "no IP addresses defined"}
2023-09-14T23:49:09Z	DEBUG	controller.windowsmachine	invalid Machine	{"name": "e2e-wm-2p7gp", "error": "no IP addresses defined"}
2023-09-14T23:49:19Z	DEBUG	controller.windowsmachine	reconciling	{"windowsmachine": {"Name":"e2e-wm-2p7gp","Namespace":"openshift-machine-api"}}
2023-09-14T23:49:34Z	INFO	controller.windowsmachine	processing	{"windowsmachine": {"Name":"e2e-wm-2p7gp","Namespace":"openshift-machine-api"}, "address": "10.0.128.8"}
```

where `e2e-wm-xx85w` was the machine configured with the outdated key, and `e2e-wm-2p7gp` is the resulting one after the reconciliation.